core/camel-api/src/main/java/org/apache/camel/ConsumerListener.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.camel;
+
+import java.util.function.Predicate;
+
+/**
+ * Proxies between the consumer within Camel to
+ */
+public interface ConsumerListener<C, P> {
+
+    /**
+     * Whether processing can resume or not
+     *
+     * @return true if it can, or false otherwise
+     */
+    void setResumableCheck(Predicate<?> afterConsumeEval);
```

Comment:
Method does not return anything. Can you update javadoc on these methods, such as tell what returning true or false means in the following methods